stl/inc/yvals_core.h

```diff
@@ -855,6 +855,42 @@
 
 // next warning number: STL4027
 
+// P0619R4 remove deprecated features from C++20
+#ifndef _HAS_REMOVED_FEATURES_CXX20
+#define _HAS_REMOVED_FEATURES_CXX20 (!_HAS_CXX20)
+#endif // _HAS_REMOVED_FEATURES_CXX20
+
+#ifndef _HAS_DEPRECATED_TYPETRAITS
+#define _HAS_DEPRECATED_TYPETRAITS (!_HAS_CXX20 || _HAS_REMOVED_FEATURES_CXX20)
+#endif // _HAS_DEPRECATED_TYPETRAITS
+
+#ifndef _HAS_DEPRECATED_SHARED_PTR_UNIQUE
+#define _HAS_DEPRECATED_SHARED_PTR_UNIQUE (!_HAS_CXX20 || _HAS_REMOVED_FEATURES_CXX20)
+#endif // _HAS_DEPRECATED_SHARED_PTR_UNIQUE
+
+#ifndef _HAS_DEPRECATED_NEGATE

```

Comment:
Naming: I recommend `NEGATORS` (following the removed section name) since it's a pair of types and functions. Additionally, `std::negate` is an existing, non-deprecated thing (the operator functor for unary minus).

This is also consistent with `_CXX17_DEPRECATE_NEGATORS` and `_SILENCE_CXX17_NEGATORS_DEPRECATION_WARNING`.

Reply:
Yep, fixed